cli/internal/run/run.go

```diff
@@ -45,70 +41,89 @@ occurred again).
-			if len(tasks) == 0 {
-				return errors.New("at least one task must be specified")
-			}
+// ExecuteRun executes the run command
+func ExecuteRun(ctx gocontext.Context, helper *cmdutil.Helper, signalWatcher *signals.Watcher, executionState *turbostate.CLIExecutionStateFromRust) error {
+	args := executionState.ParsedArgs
+	base, err := helper.GetCmdBase(args)
+	if err != nil {
+		return err
+	}
+	tasks := args.Command.Run.Tasks
+	passThroughArgs := args.Command.Run.PassThroughArgs
+	if len(tasks) == 0 {
+		return errors.New("at least one task must be specified")
+	}
+	opts, err := optsFromExecutionState(executionState)
+	if err != nil {
+		return err
+	}
 
-			_, packageMode := packagemanager.InferRoot(base.RepoRoot)
+	opts.runOpts.singlePackage = executionState.RepoState.Mode == "SinglePackage"
 
-			opts.runOpts.singlePackage = packageMode == packagemanager.Single
-			opts.runOpts.passThroughArgs = passThroughArgs
+	opts.runOpts.passThroughArgs = passThroughArgs
+	run := configureRun(base, opts, signalWatcher)
+	if err := run.run(ctx, tasks); err != nil {
+		base.LogError("run failed: %v", err)
+		return err
+	}
+	return nil
+}
 
-			run := configureRun(base, opts, signalWatcher)
-			ctx := cmd.Context()
-			if err := run.run(ctx, tasks); err != nil {
-				base.LogError("run failed: %v", err)
-				return err
-			}
-			return nil
-		},
+func optsFromExecutionState(executionState *turbostate.CLIExecutionStateFromRust) (*Opts, error) {
+	runPayload := executionState.ParsedArgs.Command.Run
+	opts := getDefaultOptions()
+	// aliases := make(map[string]string)
+	scope.OptsFromArgs(&opts.scopeOpts, &executionState.ParsedArgs)
+
+	// Cache flags
+	opts.cacheOpts.SkipFilesystem = runPayload.RemoteOnly
+	opts.cacheOpts.OverrideDir = runPayload.CacheDir
+	opts.cacheOpts.Workers = runPayload.CacheWorkers
+
+	// Runcache flags
+	opts.runcacheOpts.SkipReads = runPayload.Force
+	opts.runcacheOpts.SkipWrites = runPayload.NoCache
+
+	err := opts.runcacheOpts.SetTaskOutputMode(runPayload.OutputLogs)
+	if err != nil {
+		return nil, err
 	}
 
-	flags = cmd.Flags()
-	opts = optsFromFlags(flags)
-	return cmd
-}
+	// Run flags
+	if runPayload.Concurrency != "" {
+		concurrency, err := util.ParseConcurrency(runPayload.Concurrency)
+		if err != nil {
+			return nil, err
+		}
+		opts.runOpts.concurrency = concurrency
+	}
+	opts.runOpts.parallel = runPayload.Parallel
+	opts.runOpts.profile = runPayload.Profile
+	opts.runOpts.continueOnError = runPayload.ContinueExecution
+	opts.runOpts.only = runPayload.Only
+	opts.runOpts.noDaemon = runPayload.NoDaemon
+	opts.runOpts.singlePackage = runPayload.SinglePackage || (executionState.RepoState.Mode == "SinglePackage")
 
-func parseTasksAndPassthroughArgs(remainingArgs []string, flags *pflag.FlagSet) ([]string, []string) {
-	if argSplit := flags.ArgsLenAtDash(); argSplit != -1 {
-		return remainingArgs[:argSplit], remainingArgs[argSplit:]
+	if runPayload.Graph == _graphNoValue || runPayload.Graph == _graphTextValue {
```

Comment:
When would the Rust shim pass `runPayload.Graph == "true"`?
Also not that I think this is an issue, but if a user wanted to output a graph to a file named `stdout` this would instead output to `stdout` (not the file) correct?

Reply:
Hmm yeah good point. I think originally I had the default arg name set to `true` but now it's `stdout`. I figured out a better way of doing this by having the Rust field be an `Option<Option<String>>` where `None` is no flag passed, `Some(None)` is flag passed with no value, and `Some(Some("foo"))` is flag passed with `"foo"` as value. On the Go side that's represented as `*string` where `nil` is no flag, `""` is flag with no value, and `"foo"` is flag with `"foo"` value. It's a little opaque but I think it'll do for now and it has no magic values.